### DISCUSSION TITLE
installing pixi.js

### DISCUSSION THREAD

**Discussion by k11q** [nuxt/nuxt#16093] [Questions] [answered]
Hi I am trying to install pixi.js into my nuxt3 app and having a hard time. This is what I did:

I installed the npm package pixi.js

I create a plugin file @/plugins/pixijs.ts and filled it with this:

```
import * as PIXI from "pixi.js"

export default PIXI
```

then I run dev and get this error(without using/importing it in any component/page yet)

GET http://localhost:3000/_nuxt/@fs/Users/a/nuxt-pixi/node_modules/.vite/deps/pixi__js.js?v=c4d0beec net::ERR_ABORTED 504 (Gateway Timeout)

What am I doing wrong?

**Comment by IsraelOrtuno** [comment, marked as answer]
Plugins are auto-registered in Nuxt 3:

> All plugins in your plugins/ directory are auto-registered, so you should not add them to your nuxt.config separately.
> https://nuxt.com/docs/guide/directory-structure/plugins#plugins-directory

In order to create a Nuxt plugin it should export `defineNuxtPlugin`

```js
export default defineNuxtPlugin(nuxtApp => {
  // Doing something with nuxtApp
}
```
https://nuxt.com/docs/guide/directory-structure/plugins#creating-plugins

Inside that function is where you can run whatever you need with that library.

**Reply by k11q** [reply]
How would I use it for exporting pixi? I read that and don't understand the defineNuxtPlugin hook. Can you explain specifically for my case?

**Reply by IsraelOrtuno** [reply]
I don't really know that library but from what I see you don't need a plugin for this. You can directly import that `pixi` from any component in your application and follow `pixi` documentation to create what you need. I guess it'll look like this:

```vue
// components/PixiComponent.client.vue
<script setup>
import * as PIXI from 'pixi.js'

let app = new PIXI.Application({ width: 640, height: 360, background: '#1099bb' });

document.body.appendChild(app.view);
</script>
```

And then use `<PixiComponent />` anywhere in your app. Not sure if you can do `document.body.append` from `<script setup>` or if you have to wrap it within `onMounted`.

**Reply by k11q** [reply]
It doesnt work. I can't even run the app if I do that

**Reply by k11q** [reply]
> I don't really know that library but from what I see you don't need a plugin for this. You can directly import that `pixi` from any component in your application and follow `pixi` documentation to create what you need. I guess it'll look like this:
> 
> ```
> // components/PixiComponent.client.vue
> <script setup>
> import * as PIXI from 'pixi.js'
> 
> let app = new PIXI.Application({ width: 640, height: 360, background: '#1099bb' });
> 
> document.body.appendChild(app.view);
> </script>
> ```
> 
> And then use `<PixiComponent />` anywhere in your app. Not sure if you can do `document.body.append` from `<script setup>` or if you have to wrap it within `onMounted`.

Dont know what I did wrong, but I tried this again and it works. Thank you!
